src/EventTimer.ts

```diff
@@ -106,18 +106,21 @@ export class EventTimer {
 		};
 	}
 
-	mark(name: string): PerformanceEntry {
+	mark(name: string): void {
 		const longName = `gu.commercial.${name}`;
-		window.performance.mark(longName);
-
-		// Most recent mark with this name is the event we just created.
-		const mark = window.performance
-			.getEntriesByName(longName, 'mark')
-			.slice(-1)[0];
-		if (typeof mark !== 'undefined') {
-			this.events.push(new Event(name, mark));
+		if (
+			typeof window.performance !== 'undefined' &&
+			'mark' in window.performance
+		) {
+			window.performance.mark(longName);
+			// Most recent mark with this name is the event we just created.
+			const mark = window.performance
+				.getEntriesByName(longName, 'mark')
+				.slice(-1)[0];
```

Comment:
I assume this will always be undefined and not throw a `Uncaught TypeError: Cannot read property '0' of undefined`

Reply:
I think we are good in that case because if `getEntriesByName` returns an empty array so in the worst case this will return undefined:
`[].slice(-1)[0]` which we later test against